mmdeploy/codebase/mmflow/deploy/flow.py

```diff
@@ -0,0 +1,276 @@
+        device (str): A string specifying device type.
+    """
+
+    def __init__(self, model_cfg: mmcv.Config, deploy_cfg: mmcv.Config,
+                 device: str):
+        super().__init__(model_cfg, deploy_cfg, device)
+
+    def init_backend_model(self,
+                           model_files: Sequence[str] = None,
+                           **kwargs) -> torch.nn.Module:
+        """Initialize backend model.
+
+        Args:
+            model_files (Sequence[str]): Input model files. Default is None.
+
+        Returns:
+            nn.Module: An initialized backend model.
+        """
+        from .mmflow_model import build_mmflow_model
+        model = build_mmflow_model(
+            model_files,
+            self.model_cfg,
+            self.deploy_cfg,
+            device=self.device,
+            **kwargs)
+        return model
+
+    def init_pytorch_model(self,
+                           model_checkpoint: Optional[str] = None,
+                           **kwargs) -> torch.nn.Module:
+        """Initialize torch model.
+
+        Args:
+            model_checkpoint (str): The checkpoint file of torch model,
+                defaults to `None`.
+
+        Returns:
+            nn.Module: An initialized torch model generated by other OpenMMLab
+                codebases.
+        """
+        from mmflow.apis import init_model
+        model = init_model(self.model_cfg, model_checkpoint, self.device)
+        return model.eval()
+
+    def create_input(self,
+                     imgs: Union[str, np.ndarray],
+                     input_shape: Optional[Sequence[int]] = None,
+                     pipeline_updater: Optional[Callable] = None,
+                     **kwargs) -> Tuple[Dict, torch.Tensor]:
+        """Create input for mmflow processor.
+
+        Args:
+            imgs (str | np.ndarray): Input image(s).
+            input_shape (Sequence[int] | None): Input shape of image in
+                (width, height) format, defaults to `None`.
+            pipeline_updater (function | None): A function to get a new
+                pipeline.
+
+        Returns:
+            tuple: (data, img), meta information for the input image and input.
+        """
+        from mmflow.datasets.pipelines import Compose
+        valid = None
+        if isinstance(imgs, (list, tuple)):
+            if not isinstance(imgs[0], (np.ndarray, str)):
+                raise AssertionError('imgs must be strings or numpy arrays')
+        elif isinstance(imgs, (np.ndarray, str)):
+            imgs = [imgs]
+        else:
+            raise AssertionError('imgs must be strings or numpy arrays')
+
+        cfg = process_model_config(self.model_cfg, imgs, input_shape)
+        test_pipeline = Compose(cfg.pipeline)
+        valid_masks = []
+        data_list = []
+        if isinstance(imgs[0], np.ndarray) and isinstance(imgs[1], np.ndarray):
+            # directly add img and valid mask
+            data = dict(img1=imgs[0], img2=imgs[1])
```

Comment:
does model inputs require two images for inference? What if `imgs` has only one image or `np.ndarray`?